js/core/utils/type.d.ts

```diff
@@ -1,33 +1,33 @@
 export declare function type(object: any): string;
 
-export declare function isBoolean<T>(object: T): object is Extract<T, boolean>;
+export declare function isBoolean(object: any): object is boolean;
 
-export declare function isDate<T>(object: T): object is Extract<T, Date>;
+export declare function isDate(object: any): object is Date;
 
-export declare function isString<T>(object: T): object is Extract<T, string>;
+export declare function isString(object: any): object is string;
 
 export declare function type(object: any): 'object' | 'array' | 'string' | 'date' | 'null';
 
-export declare function isNumeric<T>(object: T): object is Extract<T, number>;
+export declare function isNumeric(object: any): object is number;
 
-export declare function isFunction<T>(object: T): object is Extract<T, Function>;
+export declare function isFunction(object: any): object is Function;
 
-export declare function isExponential<T>(object: T): object is Extract<T, number>;
+export declare function isExponential(object: any): object is number;
 
 export declare function isDefined<T>(object: T): object is NonNullable<T>;
 
-export declare function isObject<T = {}>(object: unknown): object is Extract<T, object>;
+export declare function isObject<T = {}>(object: unknown): object is object;
 
 export declare function isEmptyObject<T = {}>(object: T): boolean;
 
-export declare function isPlainObject<T>(object: T): object is Extract<T, object>;
+export declare function isPlainObject(object: any): object is { [key: string]: any };
 
 export declare function isPrimitive<T>(value: T): value is Exclude<T, Function>&Exclude<T, object>&Exclude<T, any[]>;
```

Comment:
Use union instead of `Exclude`
```suggestion
export declare function isPrimitive<T>(value: T): value is number | boolean | string | undefined | null;
```
What about Date?

https://www.typescriptlang.org/docs/handbook/basic-types.html

Reply:
https://developer.mozilla.org/en-US/docs/Glossary/Primitive